vtr_flow/arch/xilinx/7series_BRAM_DSP_carry.xml

```diff
@@ -349,6 +349,379 @@
+      <col type="EMPTY" startx="6" repeatx="11" starty="1" priority="19" />
+      <!-- Remove 3 BRAMs from the first column, 4 from all others to bring the total number of BRAM36s to 25 -->
+      <col type="EMPTY" startx="6" starty="1" incry="19" priority="21" />
+      <col type="EMPTY" startx="17" starty="1" incry="12" priority="21" />
+      <col type="EMPTY" startx="28" starty="1" incry="12" priority="21" />
+      <col type="EMPTY" startx="39" starty="1" incry="12" priority="21" />
+    </fixed_layout>
+
+    <!-- Artix-7 25k Logic Cell device.
+                    Data sheet     |   Actual
+                - 3,650 CLB slices | 1,833 CLB tiles
+                - 80 DSP slices    | 40 DSP tiles
+                - 45 BRAM blocks   | 44 BRAM tiles
+                - 150 IOs          | 19 IO tiles
+    -->
+    <fixed_layout name="XC7A25T-like" width="49" height="49">
+      <!-- 150 IOs (or around 19 tiles). Distribute them evenly around the border (5ish per side) -->
+      <perimeter type="EMPTY" priority="99" />
+      <corners type="EMPTY" priority="101" />
+      <col type="io" startx="0" starty="1" incry="11" priority="100" />
+      <col type="io" startx="48" starty="1" incry="11" priority="100" />
+      <row type="io" startx="1" starty="0" incrx="11" priority="100" />
+      <row type="io" startx="1" starty="48" incrx="12" priority="100" />
+
+      <!--Fill
+        with 'CLB'-->
+      <fill type="CLB" priority="10" />
+
+      <!-- In the actual 7-series BRAM and DSP are almost always placed 4 tiles apart from each
+        other. The actual repeatx for each of these block types is slightly different from the bellow due
+        to differences in how Xilinx lays out BRAM and DSP on the physical chip. The values given
+        bellow are the average spacing between DSP and BRAM columns.-->
+      <col type="DSP" startx="10" starty="1" repeatx="11" priority="20" />
+      <col type="EMPTY" startx="10" repeatx="11" starty="1" priority="19" />
+      <!-- Remove 2 DSPs from each column to bring the total number of DSP slices to 40 -->
+      <col type="EMPTY" startx="10" starty="1" incry="45" priority="21" />
+      <col type="EMPTY" startx="21" starty="1" incry="45" priority="21" />
+      <col type="EMPTY" startx="32" starty="1" incry="45" priority="21" />
+      <col type="EMPTY" startx="43" starty="1" incry="45" priority="21" />
+
+      <col type="BRAM" startx="6" starty="1" repeatx="11" priority="20" />
+      <col type="EMPTY" startx="6" repeatx="11" starty="1" priority="19" />
+    </fixed_layout>
+
+    <!-- Artix-7 35k Logic Cell device.
+                    Data sheet     |   Actual
+                - 5,200 CLB slices | 2,570 CLB tiles
+                - 90 DSP slices    | 45 DSP tiles
+                - 50 BRAM blocks   | 50 BRAM tiles
+                - 250 IOs          | 31 IO tiles
+    -->
+    <fixed_layout name="XC7A35T-like" width="58" height="58">
```

Comment:
This is another thing that is optional and I can make this change in the future but just to be aware, the 7-series devices are not square but rather rectangles. The ratio is on average about 0.25 (W/H) across the different chip sets. If you are only going for accurate counts of resources on the device then it doesn't really matter but worth mentioning.